From 40dda5b64d1a939a6914cbe4e18801cca8c2bf83 Mon Sep 17 00:00:00 2001
From: clownshen <clownshen@google.com>
Date: Tue, 13 Nov 2018 19:11:10 +0800
Subject: [PATCH 53/70] Workaround to hide LTE wording for O2

O2 persists to replace LTE as 4G for the summary of "4G Calling" or
hide it. Hiding it as short term fix and plan to replace LTE as 4G as
long term solution, tracing by b/119289864

Bug: 119068616
Test: Manual test
Change-Id: Ice2420c0723811b1a84992e91a85d678a7347e71
Merged-In: I8050349c79df23f14074653ba36c5016bf306ad8
---
 src/com/android/phone/MobileNetworkSettings.java | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/com/android/phone/MobileNetworkSettings.java b/src/com/android/phone/MobileNetworkSettings.java
index 6e0d4f824..f128f9a98 100644
--- a/src/com/android/phone/MobileNetworkSettings.java
+++ b/src/com/android/phone/MobileNetworkSettings.java
@@ -1181,10 +1181,17 @@ public class MobileNetworkSettings extends Activity  {
                     .getStringArray(R.array.enhanced_4g_lte_mode_title_variant);
             // Default index 0 indicates the default title string
             CharSequence enhanced4glteModeTitle = variantTitles[0];
+            CharSequence enhanced4glteModeSummary = getContext().getResources()
+                    .getString(R.string.enhanced_4g_lte_mode_summary);
             if (variant4glteTitleIndex >= 0 && variant4glteTitleIndex < variantTitles.length) {
                 enhanced4glteModeTitle = variantTitles[variant4glteTitleIndex];
+                // Workaround for b/119068616, O2 persists to replace LTE with 4G or hide it.
+                if (variant4glteTitleIndex == 2) {
+                    enhanced4glteModeSummary = null;
+                }
             }
             mButton4glte.setTitle(enhanced4glteModeTitle);
+            mButton4glte.setSummary(enhanced4glteModeSummary);
             mLteDataServicePref.setEnabled(hasActiveSubscriptions);
             Preference ps;
             ps = findPreference(BUTTON_CELL_BROADCAST_SETTINGS);
-- 
2.17.1

